### DISCUSSION TITLE
How to create seeds (not using fixtures)?

### DISCUSSION THREAD

**Discussion by pboling** [janko/rodauth-rails#269] [Q&A] [answered]
I am in a project that isn't using Rails' fixtures at all.  Instead we just create records as needed in tests (I know, we're working on improving 10 years of rot).

We have additional columns / attributes we've added to the standard accounts table / model, and one of those is `organization_id`.  We validate the presence of this attribute in `before_create_account`.

I am trying to use the `internal_request` feature:
```
> RodauthApp.rodauth.create_account(login: "my@asdf.com", password: "password1234!", organization_id: 2)
unhandled options passed to create_account: {:organization_id=>2}
eval error: There was an error creating your account ({"organization_id"=>"must not be nil"})
```

I have a few questions.

1. Why is the login param hard coded to `login` instead of accepting the override param name? (`login_param "email"`)
2. I am providing the `organization_id`, but it seems the internal request `create_account` is explicitly ignoring my additional arguments that are required to create the account.
3. Did I miss adding `organization_id` to an attribute list somewhere?  Perhaps a bit like Rails' `permit` strong params feature?

**Comment by pboling** [comment, marked as answer]
Found this discussion: https://github.com/janko/rodauth-rails/discussions/88

And inspired by that I switched back to the Rails model approach.  For me, since I have multiple customizations to the account, it looks like:
```
attributes = {id: 1, email: "blah@wah.com", user_id: 1, organization_id: 1, password: "WILD-stallions-098"}
Account.create!(status: "verified", **attributes)
```

**Reply by janko** [reply]
Yeah, I would generally recommend creating the account record directly in tests 👍🏻

As for your questions regarding the `internal_request` approach:

1. The keyword arguments that you're passing to the internal request method actually map to param *methods*, not values. So, `login_param` and `password_param` are configuration methods used for retrieving corresponding param names, so you pass `:login` and `:password` keyword arguments (you exclude the `_param` suffix). I assume the motivation behind this is to have consistent argument names, regardless of what the parameters were changed to.
2. Any keyword arguments that aren't one of the recognized options (`:session`, `:env`, `:account_id` etc.) and don't have a corresponding `*_param` method are ignored. The way to pass custom request parameters is through `:params`:

    ```rb
    RodauthApp.rodauth.create_account(login: "my@asdf.com", password: "password1234!", params: { "organization_id" => 2 })
    ```
4. Rodauth doesn't have a way of officially declaring custom request parameters, you just handle them manually. There is no mass assignment happening from parameters, so there is no need for strong params.
